rust/src/editor.rs

```diff
@@ -177,6 +178,28 @@ impl Editor {
         }
     }
 
+    fn cursor_start(&mut self) {
+        let start = self.view.sel_min() - self.col;
+        self.set_cursor(start, true);
+    }
+
+    fn cursor_end(&mut self) {
+        let current = self.view.sel_max();
+        let rope = self.text.clone();
+        let mut cursor = Cursor::new(&rope, current);
+        let mut new = current;
+        // Advance this cursor as this is considered a boundary.
+        if new == 0 {
```

Comment:
Thanks so much for the pointer. I misunderstood the `is_boundary` function. I believe it should be good to go now.